src/program/Program.js

```diff
@@ -276,37 +297,57 @@ export class Program {
                 : optimizeOrOptions
 
         const hashDependencies = options.hashDependencies ?? {}
-        const optimize = !!(options.optimize ?? false)
+        const explicitOptimize = options.optimize
+        // uses implied optimize=true if not explicitly set
+        const optimize = !!(explicitOptimize ?? true)
+        if (false == explicitOptimize && options.withAlt) {
+            console.warn(
+                "options.withAlt=true is ignored when options.optimize is explicitly disabled"
+            )
+        }
+        const withAlt =
+            false == explicitOptimize ? false : (options.withAlt ?? optimize)
 
         const ir = this.toIR({
             dependsOnOwnHash: options.dependsOnOwnHash ?? false,
             hashDependencies: hashDependencies,
             optimize: optimize
         })
 
-        const uplc = compileIR(ir, {
-            optimize: optimize,
-            parseOptions: IR_PARSE_OPTIONS,
-            optimizeOptions:
-                options.optimize && typeof options.optimize != "boolean"
-                    ? options.optimize
-                    : undefined
-        })
+        try {
+            const alt = withAlt ? this.compile({ optimize: false }) : undefined
+            // todo: re-use the IR from alt version to shorten the IR compilation for optimized version (~0.5 seconds in one sample)
+            // todo: or can we re-use some intermediate result within the IR-compilation process, to reduce overhead even further?
```

Comment:
I guess we could reuse the IR AST, which for the unoptimized alt program is immediately converted to a UplcProgram.

But yes, to be done in the future, not now